controllers/cluster.js

```diff
@@ -1,5 +1,99 @@
+const DockerWrapper = require('../lib/DockerWrapper');
+const Node = require('../server/models').Node;
+const Database = require('../server/models').Database;
+const Config = require('../server/models').Config;
+const Machine = require('docker-machine');
+const uuid = require('uuid/v4');
+const errHandler = e => console.log(e);
+
 module.exports = {
-	show(req, res) {
-    res.render('cluster');
+  list(req, res) {
+    return Node.findAll()
+      .then(nodes => {
+        if (req.accepts('html')) {
+          res.render('cluster/index', { nodes });
+        } else {
+          res.json({ nodes });
+        }
+      })
+      .catch(error => {
+        if (req.accepts('html')) {
+          res.status(400).send(error);
+        } else {
+          res.status(400).json({ error });
+        }
+      });
   },
-}
\ No newline at end of file
+
+  async create(req, res) {
+    const nodes = await Node.findAll();
+    const workerNodes = nodes.filter(node => node.manager === false);
+    const managerNode = nodes.find(node => node.manager === true);
+    const allNodesActive = nodes.every(node => node.active);
+    const accessToken = req.body.accessToken;
+
+    if (!allNodesActive) {
+      const errors = [{ message: 'Node cannot be added while cluster is updating.' }];
+      return res.status(400).send({ errors });
+    }
+
+    const name = `worker-${uuid()}`;
+    const nodeParams = {
+      name,
+      manager: false,
+      active: false
+    };
+
+    console.log('Creating node in db...');
+    Node.create(nodeParams)
+      .then(() => {
+        return res.status(202).send({ message: "Cluster scale started. (This process may take a few moments)" });
+      })
+      .then(DockerWrapper.createMachine(name, accessToken)) // TODO: This is a user input value. Sanitization?
```

Comment:
You've convinced me we're good for now. The only reason why I asked is because this `accessToken` input gets sent all the way down to the `docker-machine create` call (since as I understand it the docker machine node library just wraps the CLI).